api/krusty/mergeenvfrom_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package krusty_test
+
+import (
+	"testing"
+
+	kusttest_test "sigs.k8s.io/kustomize/api/testutils/kusttest"
+)
+
+// When patching, `envFrom` should merge the list instead of replacing it.
+func TestMergeEnvFrom(t *testing.T) {
+	th := kusttest_test.MakeHarness(t)
+	th.WriteF("/app/base/deployment.yaml", `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx
+spec:
+  template:
+    spec:
+      containers:
+        - name: nginx
+          image: image1
+          envFrom:
+            - configMapRef:
+                name: some-config
+            - configMapRef:
+                name: more-config
+`)
+	th.WriteK("/app/base", `
+resources:
+- deployment.yaml
+
+patchesStrategicMerge:
+- |-
+  apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    name: nginx
+  spec:
+    template:
+      spec:
+        containers:
+          - name: nginx
+            envFrom:
+              - configMapRef:
+                  name: another-config
+`)
+	m := th.Run("/app/base", th.MakeDefaultOptions())
```

Comment:
likewise just `"app",`

There's no overlay, so no need to add a directory level.